pkg/engine/datasource/graphql_datasource/graphql_sse_handler.go

```diff
@@ -185,6 +186,35 @@ func trim(data []byte) []byte {
 }
 
 func (h *gqlSSEConnectionHandler) performSubscriptionRequest(ctx context.Context) (*http.Response, error) {
+
+	var req *http.Request
+	var err error
+
+	// default to GET requests when POST is not specified in the SubscriptionConfiguration
+	if h.options.SSEMethod == http.MethodPost {
```

Comment:
Overall, the PR looks good to me.
But I think, as we're only supporting GET & POST, the field SSEMethod should just be a boolean that defaults to false (GET) or can be set to true (POST).
E.g. `SSEMethodPOST`.
Otherwise, you could set SSEMethod to `PUT` but it would still use `GET`, a bit confusing.